attribution/main.py

```diff
@@ -69,10 +69,13 @@ def tag_release(version: Version, message: Optional[str]) -> None:
     try:
         # XXX: This is a really hacky wall of commands
         project = Project.load()
-        version_file = Path(project.name) / "__version__.py"
-        version_file.write_text(f'__version__ = "{version}"\n')
-        sh(f"git add {version_file}")
-        sh(f"git commit -m 'Version bump v{version}'")
+
+        if project.config.get("version_file"):
+            version_file = Path(project.name) / "__version__.py"
+            version_file.write_text(f'__version__ = "{version}"\n')
+            sh(f"git add {version_file}")
+            sh(f"git commit -m 'Version bump v{version}'")
```

Comment:
Below, after writing the changelog, it attempts to run a `git commit --amend`. With this change, if version files are skipped, then that amend will modify whatever (unrelated) commit happens to be last in the project history. I would suggest moving the commit command outside of the conditional, and add the `--allow-empty` flag. This will create an empty commit that the changelog can be amended to afterwards, without needing to complicate the logic below.